src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -509,9 +505,28 @@ void TemporalMemory::activateCells(size_t activeColumnsSize,
       }
     }
   }
+  segmentsValid_ = false;
 }
 
-void TemporalMemory::activateDendrites(bool learn) {
+void TemporalMemory::activateDendrites(bool learn,
+                                       const vector<UInt> &extraActive,
+                                       const vector<UInt> &extraWinners)
+{
+  if( segmentsValid_ ) {
+    NTA_CHECK( extraActive.empty() )  << "tm.activateDendrites() already called";
```

Comment:
should we throw, or just NTA_INFO && return?

Reply:
I see. So the correct coding-pattern is: activate(), compute(), get() ...maybe mention that?